spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -615,97 +639,117 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
-                        maxshape=(recording.get_num_frames(), recording.get_num_channels())
-                    ),
-                    compression='gzip'
-                )
-            if use_timestamps:
-                nwbfile.add_acquisition(
-                    ElectricalSeries(
-                        name=es_name,
-                        data=ephys_data,
-                        electrodes=electrode_table_region,
-                        starting_time=recording.frame_to_time(0),
-                        timestamps=recording.frame_to_time(np.arange(recording.get_num_frames())),
-                        conversion=scalar_conversion,
-                        channel_conversion=channel_conversion,
-                        comments='Generated from SpikeInterface::NwbRecordingExtractor',
-                        description=metadata['Ecephys']['ElectricalSeries'].get('description', defaults['description'])
-                    )
-                )
-            else:
-                nwbfile.add_acquisition(
-                    ElectricalSeries(
-                        name=es_name,
-                        data=ephys_data,
-                        electrodes=electrode_table_region,
-                        starting_time=recording.frame_to_time(0),
-                        rate=recording.get_sampling_frequency(),
-                        conversion=scalar_conversion,
-                        channel_conversion=channel_conversion,
-                        comments='Generated from SpikeInterface::NwbRecordingExtractor',
-                        description=metadata['Ecephys']['ElectricalSeries'].get('description', defaults['description'])
-                    )
+        eseries_kwargs = dict(
+            name=es_name,
+            data=H5DataIO(ephys_data, compression="gzip"),
+            electrodes=electrode_table_region,
+            conversion=scalar_conversion,
+            channel_conversion=channel_conversion,
+            comments="Generated from SpikeInterface::NwbRecordingExtractor"
+        )
+        if not use_timestamps:
+            eseries_kwargs.update(
+                starting_time=recording.frame_to_time(0),
+                rate=recording.get_sampling_frequency()
+            )
+        else:
+            eseries_kwargs.update(
+                timestamps=H5DataIO(
+                    recording.frame_to_time(np.arange(recording.get_num_frames())),
+                    compression="gzip"
                 )
+            )
+
+        if not write_as_lfp:
+            eseries_kwargs.update(
+                description=metadata['Ecephys']['ElectricalSeries'].get('description', raw_defaults['description'])
+            )
+            nwbfile.add_acquisition(ElectricalSeries(**eseries_kwargs))
+        else:
+            eseries_kwargs.update(
+                description=metadata['Ecephys']['ElectricalSeries'].get('description', lfp_defaults['description'])
```

Comment:
I think the description in this case should in in Ecephys.LFP

Reply:
Good  catch!